Sources/Fuzzilli/Evaluation/ProgramCoverageEvaluator.swift

```diff
@@ -15,29 +15,61 @@
 import Foundation
 import libcoverage
 
-class CovEdgeSet: ProgramAspects {
-    let count: UInt64
-    let edges: UnsafeMutablePointer<UInt32>?
+public class CovEdgeSet: ProgramAspects {
+    var count: UInt64
+    var edges: UnsafeMutablePointer<UInt32>?
 
     init(edges: UnsafeMutablePointer<UInt32>?, count: UInt64) {
         self.count = count
         self.edges = edges
         super.init(outcome: .succeeded)
     }
-    
+
     deinit {
         free(edges)
     }
 
-    override var description: String {
+    public override var description: String {
         return "new coverage: \(count) newly discovered edge\(count > 1 ? "s" : "") in the CFG of the target"
     }
+
+    /// This adds additional copies, but is only hit when new programs are added to the corpus
+    /// It is used by corpus schedulers such as MarkovCorpus that require knowledge of which samples trigger which edges
+    public func toEdges() -> [UInt32] {
+        return Array(UnsafeBufferPointer(start: edges, count: Int(count)))
+    }
+
+    public static func == (lhs: CovEdgeSet, rhs: CovEdgeSet) -> Bool {
+        if lhs.outcome != rhs.outcome { return false }
+        if lhs.count != rhs.count { return false }
+        for i in 0..<Int(lhs.count) {
+            if lhs.edges![i] != rhs.edges![i] {
+                return false
+            }
+        }
+        return true
+    }
+
+    func setEdges<T: Collection>(_ collection: T) where T.Element == UInt32 {
+        // Update internal state to match the intersection
```

Comment:
Nit: I think this comment is outdated

Reply:
Also I think there should be a `precondition` here that `self.count >= collection.count` because I think otherwise memory would be corrupted

Reply:
And maybe let's make this method `fileprivate` if possible to reflect the fact that it is an internal method and shouldn't be used anywhere else